wandb/sdk/wandb_run.py

```diff
@@ -84,6 +86,35 @@ def exc_handler(self, exc_type, exc, *tb):
         traceback.print_exception(exc_type, exc, *tb)
 
 
+class RunStatusChecker(object):
+    """Periodically polls the background process for relevant updates.
+
+    For now, we just use this to figure out if the user has requested a stop.
+    """
+
+    def __init__(self, interface, polling_interval=15):
+        self._interface = interface
+        self._polling_interval = polling_interval
+
+        self._join_event = threading.Event()
+        self._thread = threading.Thread(target=self.check_status)
+        self._thread.daemon = True
+        self._thread.start()
+
+    def check_status(self):
+        join_requested = False
+        while not join_requested:
+            status_result = self._interface.send_status_sync(check_stop_req=True)
```

Comment:
If this times out `status_result` will be None.  We should probably just ignore it if it's None for now.

Reply:
True, good catch. Actually now that I'm looking at this more, I realize we're not supposed to send these kinds of sync requests from background threads...I need to think more about this.